falak/pkg/api/calendar/service.go

```diff
@@ -289,11 +292,97 @@ func (s *service) GetCalendarsWithCalendarAccounts(ctx context.Context, r *conne
 		},
 	}, nil
 }
+func (s *service) SchedulePrayerTimes(ctx context.Context, r *connect.Request[calendarv1.SchedulePrayerTimesRequest]) (*connect.Response[calendarv1.SchedulePrayerTimesResponse], error) {
+	if err := s.pv.Validate(r.Msg); err != nil {
+		log.Ctx(ctx).Err(err).Msg("invalid request")
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	ipAddr := r.Header().Get("CF-IP")
+	if ipAddr == "" {
+		log.Ctx(ctx).Error().Msg("we didn't receive an IP address")
+		return nil, internalError
+	}
+
+	geoLocResp, err := s.geoLocationClient.GetGeoLocationInfo(ctx, &geolocationclient.GetGeoLocationInfoRequest{
+		Ip: ipAddr,
+	})
+	if err != nil {
+		log.Ctx(ctx).Err(err).Msg("failed to retrieve location from IP address")
+		return nil, internalError
+	}
+
+	fmt.Println(geoLocResp)
+	fmt.Println(time.Now().Format("02-01-2006"))
+
+	prayerTimes, err := s.prayerClient.GetPrayertimeInfo(ctx, &prayerclient.GetPrayertimeInfoRequest{
+		Date:    time.Now().Format("02-01-2006"),
+		City:    geoLocResp.City,
+		Country: geoLocResp.Country,
+	})
+	if err != nil {
+		log.Ctx(ctx).Err(err).Msg("failed to retrieve prayer times from Adhan API")
+		return nil, internalError
+	}
+
+	fmt.Printf("prayer times: %v\n", prayerTimes)
```

Comment:
remove this part, and do the following:
1. send back this `https://prayerwebcal.dsultan.com/loc/Riyadh_Saudi_Arabia`
2. Riyadh_Saudi_Arabia is made like this: `CITYNAME_COUNTRYNAME`, if any space exist, replace them with `_`
```
return &connect.Response[calendarv1.SchedulePrayerTimesResponse]{
		Msg: &calendarv1.SchedulePrayerTimesResponse{
// put some stuff here, the response url, so the client can use it
                  },
	}, nil
```